src/pollers/gateway.rs

```diff
@@ -54,6 +59,9 @@ pub struct ServerGatewayOptions {
     /// attempt to use TLS when connecting to the Temporal server. Lang SDK is expected to pass any
     /// certs or keys as bytes, loading them from disk itself if needed.
     pub tls_cfg: Option<TlsConfig>,
+
+    /// Optional retry configuration for the server gateway. If not supplied defaults will be used.
+    pub retry_config: Option<RetryConfig>,
```

Comment:
I'd say there's no reason to make this an `Option` -- it's trivial for the caller to use `Default` if they don't want to specify some or all of the specific fields.

Reply:
The counter argument would be that it's simpler to set None and then it's as simple for us to default it down the line.

Reply:
I'd avoid it. It's not a standard practice, and it results in dumb things happening with more than one layer. EX: if you apply default to *this* struct the field will be `None`. It would make more sense for it to be populated with the defaults.